File: NYCTaxiCabApp.Web/ClientApp/src/app/models/ride.ts
import { Time } from '../models/time';

export interface IRide{
    distanceCoveredSlowSpeed: number;
    distanceCoveredFasterSpeed: number;
    durationWithSlowSpeed: Time;
    durationWithFastSpeed: Time; 
    amountTimeWithoutMotion: Time;
    rideDate: Date; 
    rideTime: Time; 
    originAddress: string; 
    destinationAdress: string; 
    isInValid(): boolean;
} 

export class Ride implements IRide{
    distanceCoveredSlowSpeed: number;
    distanceCoveredFasterSpeed: number;
    durationWithSlowSpeed: Time;
    durationWithFastSpeed: Time;
    amountTimeWithoutMotion: Time;
    rideDate: Date;
    rideTime: any;
    originAddress: string;
    destinationAdress: string;

    public isInValid(): boolean{
        let distSlow = this.distanceCoveredSlowSpeed || 0;
        
        return (distSlow + this.getMinutes(this.durationWithFastSpeed) + this.getMinutes(this.amountTimeWithoutMotion)) <= 0 ;
    }

    private getMinutes(time: Time): number{
        if(time == undefined || time == null)
            return 0;
        return time.hour * 60 + time.minute;
    }

}

File: NYCTaxiCabApp.Web/Models/Ride.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace NYCTaxiCabApp.Web.Models
{
    public class Ride
    {
        public double DistanceCoveredSlowSpeed { get; set; }
        public double DistanceCoveredFasterSpeed { get; set; }
        public Time DurationWithSlowSpeed { get; set; }
        public Time DurationWithFastSpeed { get; set; }
        public Time AmountTimeWithoutMotion { get; set; }
        public Date RideDate { get; set; } 
        public Time RideTime { get; set; }
        public string OriginAddress { get; set; } 
        public string DestinationAdress { get; set; } 
    }
}
